src/main/java/net/fabricmc/fabric/api/server/PlayerLookupHelper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.server;
+
+import net.fabricmc.fabric.impl.server.PlayerLookupHelperImpl;
+import net.fabricmc.loader.FabricLoader;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.ChunkPos;
+
+import java.util.stream.Stream;
+
+public interface PlayerLookupHelper {
```

Comment:
To me, `lookup` implies "looking up the player's Mojang account info" for stuff like skins. Maybe this could be named something like `PlayerSearch` or `PlayerSelector`?